src/main/java/com/commercetools/pspadapter/notification/processor/impl/NotificationProcessorBase.java

```diff
@@ -54,17 +50,15 @@
                 );
     }
 
-    protected ImmutableList<UpdateAction<Payment>> createPaymentUpdates(@Nonnull Payment ctpPayment,
+    protected List<UpdateAction<Payment>> createPaymentUpdates(@Nonnull Payment ctpPayment,
                                                                         @Nonnull Event event) {
-        final ImmutableList.Builder<UpdateAction<Payment>> listBuilder = ImmutableList.builder();
-        listBuilder.add(createAddInterfaceInteractionAction(event));
-        Optional<ChangeTransactionState> changeTransactionOpt = createChangeTransactionState(ctpPayment);
-        if (changeTransactionOpt.isPresent()) {
-            listBuilder.add(changeTransactionOpt.get());
-        } else {
-            logger.warn("Notification event {} did not trigger change transaction state", event);
+        ArrayList<UpdateAction<Payment>> updateActions = new ArrayList<>();
+        updateActions.add(createAddInterfaceInteractionAction(event));
+        List<? extends UpdateAction<Payment>> updateAtions = createChangeTransactionState(ctpPayment);
+        if (!updateAtions.isEmpty()) {
+            updateActions.addAll(updateAtions);
         }
-        return listBuilder.build();
+        return updateActions;
     }
 
     @SuppressWarnings("unchecked")
```

Comment:
I missed it in previous commits, but it is never late to fix it:
pls, avoid _SuppressWarnings_ as much as possible! Make safe casting and log warning (error) in case of unexpected input type.

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/58/commits/9348593340ed30589d914929898313623a9ae732

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/commit/24f840caf37122ca485127ab418ebd313d0090e8#diff-3040f68b522661b2bb0b5439a80fe434L64